server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -84,6 +83,27 @@ public Boolean isAcquired(LockInfo lockInfo) throws IOException {
         return !lockFiles.isEmpty();
     }
 
+    /**
+     * Acquires lock on the file mentioned in originalLockInfo for acquirer mentioned in clonedLockInfo.
+     * Snapshot layer enforces thread safety by having checks in place to ensure that the source snapshot is not being deleted before proceeding
+     * with the clone operation. Hence, the original lock file would always be present while acquiring the lock for cloned snapshot.
+     * @param originalLockInfo lock info instance for original lock.
+     * @param clonedLockInfo lock info instance for which lock needs to be cloned.
+     * @throws IOException throws IOException if originalResource itself do not have any lock.
+     */
+    @Override
+    public void cloneLock(LockInfo originalLockInfo, LockInfo clonedLockInfo) throws IOException {
+        assert originalLockInfo instanceof FileLockInfo : "originalLockInfo should be instance of FileLockInfo";
+        assert clonedLockInfo instanceof FileLockInfo : "clonedLockInfo should be instance of FileLockInfo";
+        String originalResourceId = Objects.requireNonNull(((FileLockInfo) originalLockInfo).getAcquirerId());
+        String clonedResourceId = Objects.requireNonNull(((FileLockInfo) clonedLockInfo).getAcquirerId());
```

Comment:
I would remove this method and instead create a `clone()` inside `LockInfo` that returns a clone of the originalLockInfo, then override the acquirer and then make a direct `acquireLock` call.

Reply:
`LockInfo` is a very thin interface , so we wouldn't be able to clone it .  We can take a relook on this after Sachin's refactor of LockManager : https://github.com/opensearch-project/OpenSearch/pull/8399